code/controllers/subsystem/job.dm

```diff
@@ -590,14 +603,190 @@ SUBSYSTEM_DEF(job)
+				file_data["[job_key]"] = list(
+					"#Total Positions" = -1,
+					"#Spawn Positions" = -1,
+					"#Playtime Requirements" = occupation.exp_requirements,
+					"#Required Account Age" = occupation.minimal_player_age,
+				)
+				continue
+			// Generate new config from codebase defaults.
+			file_data["[job_key]"] = list(
+				"#Total Positions" = occupation.total_positions,
+				"#Spawn Positions" = occupation.spawn_positions,
+				"#Playtime Requirements" = occupation.exp_requirements,
+				"#Required Account Age" = occupation.minimal_player_age,
+			)
+		if(!export_toml(user, file_data))
+			return FALSE
+		return TRUE
+
+/// If we add a new job or more fields to config a job with, quickly spin up a brand new config that inherits all of your old settings, but adds the new job with codebase defaults. Returns TRUE if a file is successfully generated, FALSE otherwise.
+/datum/controller/subsystem/job/proc/regenerate_job_config(mob/user)
+	var/toml_file = "[global.config.directory]/jobconfig.toml"
+	var/list/file_data = list()
+
+	if(!fexists(file(toml_file))) // You need an existing (valid) TOML for this to work. Sanity check if someone calls this directly instead of through 'Generate Job Configuration' verb.
+		to_chat(user, span_notice("No jobconfig.toml found in the config folder! If this is not expected, please notify a server operator or coders. You may need to generate a new config file by running 'Generate Job Configuration' from the Server tab."))
+		return FALSE
+
+	var/job_config = rustg_read_toml_file(toml_file)
+	for(var/datum/job/occupation as anything in joinable_occupations)
+		var/job_name = occupation.title
+		var/job_key = occupation.config_tag
+
+		var/default_positions = job_config[job_key]["Total Positions"]
+		var/starting_positions = job_config[job_key]["Spawn Positions"]
+		var/playtime_requirements = job_config[job_key]["Playtime Requirements"]
+		var/required_account_age = job_config[job_key]["Required Account Age"]
+
+		// When we regenerate, we want to make sure commented stuff stays commented, but we also want to migrate information that remains uncommented. So, let's make sure we keep that pattern.
+		if(job_config["[job_key]"]) // Let's see if any data for this job exists.
+			if(file_data["[job_key]"]) // Sanity, let's just make sure we don't overwrite anything or add any dupe keys.
+				continue
+			if(default_positions) // If the variable exists, we want to ensure it migrated into the new TOML uncommented, to allow for flush migration.
+				file_data["[job_key]"] += list(
+					"Total Positions" = default_positions,
+				)
+			else // If we can't find anything for this variable, then we just throw in the codebase default with it commented out.
+				file_data["[job_key]"] += list(
+					"#Total Positions" = occupation.total_positions,
+				)
+
+			if(starting_positions) // Same pattern as above.
+				file_data["[job_key]"] += list(
```

Comment:
Why `+= list(` instead of just setting the field directly?

`file_data["[job_key]"]["Spawn Positions"] = starting_positions`

Reply:
I was getting some nasty runtimes for some reason last i recall (it's been a second since i've thought, i'll try it again later today

Reply:
Yeah, found what I kept running into:

![image](https://user-images.githubusercontent.com/34697715/196101686-6a464c43-e5c2-47a0-955e-e3a80d90e2af.png)

![image](https://user-images.githubusercontent.com/34697715/196101709-b399f893-c915-4e71-8924-45ff1fd651cb.png)

It works fine if you do it how I have it on current commit (combining lists), so... I'm not really sure what's going on there.

Reply:
bad index on string accesses means the list is null.

Reply:
Okay then this is fine, I just didn't know it could be